### PR TITLE
Bug fix of memory overflow when calculating model deviation

### PR DESCRIPTION
**Pull request by Ericwang6**
Calculate model deviation frame by frame to avoid memory overflow problems

### CODE REVIEW

File: deepmd/infer/model_devi.py [hunk 1]
```diff
@@ -195,12 +195,13 @@ def make_model_devi(
...
     for data in data_sets:
-        coord = data["coord"]
-        box = data["box"]
-        atype = data["type"][0] 
-        devi = calc_model_devi(coord, box, atype, dp_models, nopbc=nopbc)
-        nframes_tot += coord.shape[0]
-        devis.append(devi)
+        coords = data["coord"]
+        boxs = data["box"]
+        atype = data["type"][0]
```
Comment (njzjz) [inline deepmd/infer/model_devi.py:200]: Is there a reason to only use the first (?) type?

Comment (Ericwang6) [reply deepmd/infer/model_devi.py:200]: I think atom types for all frames in a dataset is the same, aren't they?

Comment (njzjz) [reply deepmd/infer/model_devi.py:200]: I don't think we can publicly ensure it if the returned array is the size of `nframes*natoms` instead of `natoms`

Comment (amcadmus) [reply deepmd/infer/model_devi.py:200]: I agree with @njzjz , in this context we do not assume all frames of data["type"] has the same value.

**Review by amcadmus — CHANGES_REQUESTED** [review]

**Comment by amcadmus** [comment]
Next time you PR,  please checkout a new branch. 
your devel has too many commits that do not exist on the current devel branch.

**Comment by amcadmus** [comment]
fixed by #1165

**Comment by njzjz** [comment]
> fixed by #1165

I don't think so... `dp test` and `dp model-devi` are two different entry points.

**Comment by amcadmus** [comment]
@njzjz It should be straightforward of using `AutoBatchSize` here, what do you think?

**Review by amcadmus — APPROVED** [review]

**Comment by njzjz** [comment]
> @njzjz It should be straightforward of using `AutoBatchSize` here, what do you think?

Agree, and I've not decided whether we should use it on DeepPot -- the Python interface provided to users.

**Review by njzjz — APPROVED** [review]

**Comment by amcadmus** [comment]
> > @njzjz It should be straightforward of using `AutoBatchSize` here, what do you think?
> 
> Agree, and I've not decided whether we should use it on DeepPot -- the Python interface provided to users.

Good idea!

**Pull request merged**